README.md

```diff
@@ -67,3 +67,48 @@ jobs:
           chmod +x dist/bin/*
           dist/bin/${{ matrix.bin }} --version
 ```
+
+## Workflows
+
+### [`haskell-ci`](./.github/workflows/haskell-ci.yml)
+
+Build and test single-package Haskell projects using Cabal. Performs a sequence
+of steps that are appropriate for many projects, namely:
+
+- Uses `haskell-actions/setup` to install Cabal and GHC
+- Runs `cabal sdist`, unpack the `sdist`, and perform subsequent steps there
+- Uses `actions/cache` to cache the Cabal store and build directory (`dist-newstyle`)
+- Runs `cabal configure`, `build`, `test`, `sdist`, `check`, and `haddock`
+- Unconditionally saves the cache
+- Checks that the package is compatible with GHC's bundled version of Cabal
+
+This workflow will not be applicable to all projects, but it is applicable to
+enough to be useful.
+
+For a description of the inputs, see the workflow itself.
+
+Example:
+
+```yml
+name: CI
+on:
+  push:
+    branches: [main]
+  pull_request:
+  workflow_dispatch:
+jobs:
+  ci:
+    name: ${{ matrix.os }} GHC-${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-24.04]
+        ghc: [9.8.4, 9.10.1, 9.12.1]
+      fail-fast: false
+    uses:
+      haskell-ci:
+        uses: GaloisInc/.github/workflows/haskell-ci@<SHA>
+        inputs:
+          ghc: ${{ matrix.ghc }}
+          os: ${{ matrix.os }}
```

Comment:
Food for thought: for the sake of extensibility, does it make sense to have a notion of a "hooks" for this workflow? That is, allow users to specify shell commands to run before (i.e., a pre-hook) or after (i.e., a post-hook) the Haskell-related parts of the workflow. You could imagine something like this, inspired by [`grease`'s CI](https://github.com/GaloisInc/grease/blob/db58869d7096da9814e18a3c8f8c4955ddea304b/.github/workflows/docker.yml#L28-L30):

```yaml
    uses:
      haskell-ci:
        uses: GaloisInc/.github/workflows/haskell-ci@v1
        inputs:
          ghc: ${{ matrix.ghc }}
          os: ${{ matrix.os }}
          pre-hook: |
            sed -i 's/VERSION/0.1/' versioninfo.txt
```

Reply:
Yes, I think that's a good idea. Mind if we defer to an issue? I think there's plenty of repos that won't need this, so I wouldn't mind making incremental progress here.

Reply:
I'm fine with deferring this to an issue.

Reply:
https://github.com/GaloisInc/.github/issues/17